thunder/tests/opinfos.py

```diff
@@ -6152,6 +6152,32 @@ def sample_generator(op, device, dtype, requires_grad, **kwargs):
 nn_ops.append(max_pool3d_opinfo)
 
 
+def one_hot_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=torch.long, requires_grad=requires_grad)
```

Comment:
`dtype=dtype` and let's restrict the generation dtype to be `torch.long` elsewhere